UnitySetup/UnitySetup.psm1

```diff
@@ -2616,23 +2612,29 @@ function Get-ScopedRegistry {
 function Import-UnityProjectManifest {
     [CmdletBinding()]
     param(
-        [Parameter()]
-        [String]$ProjectManifestPath,
-        [String]$SearchPath,
-        [int]$SearchDepth = 3
-    )
+        [ValidateScript({if(-not [string]::IsNullOrEmpty($_)) { Test-Path $_ -PathType Leaf }}, ErrorMessage = "`"{0}`" is not a valid file")]
+        [ValidateNotNullOrEmpty()]
+        [Parameter(Mandatory=$true, ParameterSetName = "ProjectManifest")]
+        [Parameter(Mandatory=$true, ParameterSetName = "SearchPathAndProjectManifest")]
+        [string]$ProjectManifestPath,
 
-    if (-not $PSBoundParameters.ContainsKey('SearchPath') -and $PSBoundParameters.ContainsKey('ProjectManifestPath')) {
-        Write-Error "A SearchPath or ProjectManifestPath must be provided."
-    }
+        [ValidateNotNullOrEmpty()]
+        [Parameter(Mandatory=$true, ParameterSetName = "SearchPath")]
+        [Parameter(Mandatory=$true, ParameterSetName = "SearchPathAndProjectManifest")]
+        [string]$SearchPath,
+
+        [Parameter(Mandatory=$false, ParameterSetName = "SearchPath")]
+        [Parameter(Mandatory=$false, ParameterSetName = "SearchPathAndProjectManifest")]
+        [uint]$SearchDepth = 3
+    )
```

Comment:
I'm not familiar with these validation attributes.  Since we're moving towards the use of either a `$ProjectManifestPath` OR `$SearchPath` + `$SearchDepth`  I am assuming that you can create different validation rules for different Parameter sets?

Does this set of rules express that if `$ProjectManifest` is provided, then `$SearchPath and $SearchDepth` become mandatory, while if the search parameters are provided, the manifest parameter must be null?

Reply:
Currently what I have requires that you specify either `ProjectManifestPath` or `SearchPath` or both, and it ensures that you cannot pass `SearchDepth` unless `SearchPath` is provided. The logic seemed to be that both was allowed, but we can make it be strict 'or' if that's the behavior we want.